components/ClassInviteTable.js

```diff
@@ -22,21 +33,67 @@ export default function ClassInviteTable({ classes }) {
 
   const deleteClass = async () => {
     if (confirm('Do you want to delete this class?') == true) {
-      const response = await fetch(`/api/deleteclass`, {
-        method: 'DELETE',
-        body: JSON.stringify(classes.classroomId)
-      });
-      router.reload('http://localhost:3000/classes');
-      alert('Successfully Deleted Class');
-      return await response.json();
+      const JSONdata = JSON.stringify(classes.classroomId);
+      try {
+        const res = await fetch(`/api/deleteclass`, {
+          method: 'DELETE',
+          headers: {
+            'Content-Type': 'application/json'
+          },
+          body: JSONdata
+        });
+        if (res.status === 403) {
+          router.reload('http://localhost:3000/classes');
+          alert('Cannot delete class, not valid user');
```

Comment:
nit: specify that the current user does not own the class rather than saying the user is not valid